hoot-core/src/main/cpp/hoot/core/conflate/frechet/FrechetDistance.cpp

```diff
@@ -215,8 +215,8 @@ Radians FrechetDistance::getHeadingAvg(WayPtr way, int index)
   if (start < 0)
     start = 0;
   //  This case is a way with only one point
-  if (stop == 0)
-    return WayHeading::calculateHeading(WayLocation(_map, way, 0, 0), 0.5);
+  if (start == stop)
+    return WayHeading::calculateHeading(WayLocation(_map, way, start, 0), 0.5);
```

Comment:
It is possible (thought very difficult) to get passed this and iterate the loop below 0 times and therefore get a divide-by-zero condition so this change ensures that cannot happen.